execute/plugin.go

```diff
@@ -240,11 +243,35 @@ func (p *Plugin) Observation(
 		}
 
 		// TODO: Fire off messages for an attestation check service.
-		return exectypes.NewObservation(groupedCommits, messages).Encode()
+		return exectypes.NewObservation(groupedCommits, messages, nil).Encode()
 
 	case exectypes.Filter:
-		// TODO: pass the previous two through, add in the nonces.
-		return types.Observation{}, fmt.Errorf("unknown state")
+		// TODO: add in nonces, other data comes from previous outcome.
+		nonceRequestArgs := make(map[cciptypes.ChainSelector]map[string]struct{})
+
+		// Collect unique senders.
+		for _, commitReport := range previousOutcome.Report.ChainReports {
+			if _, ok := nonceRequestArgs[commitReport.SourceChainSelector]; !ok {
+				nonceRequestArgs[commitReport.SourceChainSelector] = make(map[string]struct{})
+			}
+			for _, msg := range commitReport.Messages {
+				nonceRequestArgs[commitReport.SourceChainSelector][msg.Sender.String()] = struct{}{}
```

Comment:
I think this should use `typeconv.AddressBytesToString` so that the address is encoded in the chain-family-native string encoding. `String()` here always returns hex.

Reply:
You're right, we'll want it in the native format when looking up the value from the reader.